src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs

```diff
@@ -1790,11 +1790,18 @@ private string ReadLineFromConsole(bool endOnTab, string initialContent, bool ca
                         continue;
                     }
 
+                    // Handle case where terminal gets reset and the index is outside of the buffer
+                    if (index > s.Length)
+                    {
+                        index = s.Length;
+                    }
+
                     // Modify string
                     if (!insertMode) // then overwrite mode
                     {
                         s = s.Remove(index, 1);
```

Comment:
When `index == s.Length`, this line will throw. The if check probably should be updated to `if (!insertMode && index < s.Length)`